### ISSUE TITLE
Dombus fails after Domoticz transfer to Docker

### ISSUE THREAD

**Issue by daserra23** [CreasolTech/CreasolDomBus#4]
I have moved domoticz to docker and all it's plugins and scripts in the "config" folder.
My docker-compose.yml is
`version: '3'
services:
  domoticz:
    image: domoticz/domoticz:stable
    container_name: domoticz
    restart: unless-stopped
    network_mode: host
    devices:
      - "/dev/serial/by-id/usb-0658_0200-if00:/dev/ttyACM0"
      - "/dev/serial/by-id/usb-1a86_USB2.0-Ser_-if00-port0:/dev/ttyUSB0"
      - "/dev/serial/by-id/usb-RFXCOM_RFXtrx433_A11P1S9K-if00-port0:/dev/ttyUSB1"
      - "/dev/serial/by-id/usb-Silicon_Labs_slae.sh_cc2652rb_stick_-_slaesh_s_iot_stuff_00_12_4B_00_22_98_90_8F-if00-port0:/dev/ttyUSB2"
    ports:
      - "8080:8080"
    volumes:
      - /home/pi/domoticzDocker/config:/opt/domoticz/userdata
    environment:
      - TZ=Europe/Lisbon`

I am getting these errors

` 2024-11-30 14:51:39.410 Error: Dombus: Call to function 'onStart' failed, exception details:
2024-11-30 14:51:39.418 Error: Dombus: Traceback (most recent call last):
2024-11-30 14:51:39.418 Error: Dombus: File "/opt/domoticz/userdata/plugins/CreasolDomBus/CreasolDomBusProtocol.py", line 249, in portsDisabledInit
2024-11-30 14:51:39.418 Error: Dombus: fd=open(PORTSDISABLED)
2024-11-30 14:51:39.418 Error: Dombus: FileNotFoundError: [Errno 2] No such file or directory: 'plugins/CreasolDomBus/portsDisabled.json'
2024-11-30 14:51:39.418 Error: Dombus:
2024-11-30 14:51:39.418 Error: Dombus: During handling of the above exception, another exception occurred:
2024-11-30 14:51:39.418 Error: Dombus:
2024-11-30 14:51:39.418 Error: Dombus: Traceback (most recent call last):
2024-11-30 14:51:39.418 Error: Dombus: File "/opt/domoticz/userdata/plugins/CreasolDomBus/plugin.py", line 171, in onStart
2024-11-30 14:51:39.418 Error: Dombus: _plugin.onStart()
2024-11-30 14:51:39.418 Error: Dombus: File "/opt/domoticz/userdata/plugins/CreasolDomBus/plugin.py", line 82, in onStart
2024-11-30 14:51:39.418 Error: Dombus: dombus.portsDisabledInit()
2024-11-30 14:51:39.418 Error: Dombus: File "/opt/domoticz/userdata/plugins/CreasolDomBus/CreasolDomBusProtocol.py", line 254, in portsDisabledInit
2024-11-30 14:51:39.418 Error: Dombus: portsDisabledWriteNow()
2024-11-30 14:51:39.418 Error: Dombus: File "/opt/domoticz/userdata/plugins/CreasolDomBus/CreasolDomBusProtocol.py", line 242, in portsDisabledWriteNow
2024-11-30 14:51:39.418 Error: Dombus: with open(PORTSDISABLED, 'w') as fd:
2024-11-30 14:51:39.418 Error: Dombus: FileNotFoundError: [Errno 2] No such file or directory: 'plugins/CreasolDomBus/portsDisabled.json' `

Inside the container the files exist

`ls -l /opt/domoticz/userdata/plugins/CreasolDomBus/portsDisabled.json
-rw-rw-rw- 1 1000 1000 2 Mar 28  2023 /opt/domoticz/userdata/plugins/CreasolDomBus/portsDisabled.json
`
I did try mapping "/dev/serial/by-id/usb-0658_0200-if00:/dev/serial/by-id/usb-0658_0200-if00" but Dombus couldnt see that despite it being in /dev/serial/by-id inside the container. Not sure if this is relevent.

Would be nice to get the hardware working with Domoticz in the container.

**Comment by daserra23** [comment]
Forget about this, domoticz on docker isn't for me.

**Comment by CreasolTech** [comment]
Hi Davide,
dombus plugins opens file "plugins/CreasolDomBus/portsDisabled.json" : maybe the cwd of the docker environment is different.
It's good to know, for me, if changing the portsDisabled path to "userdata/plugins/CreasolDomBus/portsDisabled.json" solve the problem.
Regarding the serial device, you have to enabled it in the docker configuration.
Thanks.
Paolo

**Comment by daserra23** [comment]
The domoticz docker set up is a mess, the config files move in and out of userdata so sometimes your plugin exists, sometimes it doesn't, ports are not mapped properly and I have spent too much time on it already. I'll be running domoticz in a proxmox vm instead, this is the second time in a year I have tried to get this going and failed and I've wasted many hours on it. It's not your problem, it's the sad state of domoticz documentation and forum support. I look forward to continuing to use Dombus, thanks for your support.
